source/MaterialXCore/Interface.h

```diff
@@ -201,9 +203,35 @@ class Input : public PortElement
     }
 
     /// @}
+    /// @name Default Geometric Property
+    /// @{
+
+    /// Set the defaultgeomprop string for the input.
+    void setDefaultGeomPropString(const string& geomprop)
+    {
+        setAttribute(DEFAULTGEOMPROP, geomprop);
+    }
+
+    /// Return true if the given input has a defaultgeomprop string.
+    bool hasDefaultGeomPropString() const
+    {
+        return hasAttribute(DEFAULTGEOMPROP);
+    }
+
+    /// Return the defaultgeomprop string for the input.
+    const string& getDefaultGeomPropString() const
+    {
+        return getAttribute(DEFAULTGEOMPROP);
+    }
+
+    /// Return the GeomPropDef element to use, if defined for this input.
+    GeomPropDefPtr getDefaultGeomProp() const;
+
+    /// @}
 
   public:
     static const string CATEGORY;
+    static const string DEFAULTGEOMPROP;
```

Comment:
For consistency with other MaterialX code, the name of this string constant should be DEFAULT_GEOM_PROP_ATTRIBUTE.